packages/blade/src/components/Accordion/__tests__/Accordion.native.test.tsx

```diff
@@ -0,0 +1,195 @@
+
+    // Clicking second accordion item
+    const trigger2 = getAllByRole('button')[1];
+    expect(trigger2).toHaveAccessibilityState({ expanded: false });
+    expect(getByText(description2)).not.toBeVisible();
+
+    fireEvent.press(trigger2);
+    // wait for animation to finish
+    await waitFor(() => expect(getByText(description2)).toBeVisible());
+
+    expect(trigger2).toHaveAccessibilityState({ expanded: true });
+    expect(trigger1).toHaveAccessibilityState({ expanded: false });
+    expect(getByText(description1)).not.toBeVisible();
+
+    expect(onExpandChange).toHaveBeenCalledTimes(2);
+    expect(onExpandChange).toHaveBeenLastCalledWith({ expandedIndex: 1 });
+
+    mockConsoleError.mockRestore();
+  });
+
+  it('should expand and close in controlled behavior', async () => {
+    // Suppress an act warning that is likely fixed in recent versions https://github.com/callstack/react-native-testing-library/issues/379
+    const mockConsoleError = jest.spyOn(console, 'error').mockImplementation();
+
+    const button1 = 'How can I setup Route?';
+    const description1 =
+      'You can use Razorpay Route from the Dashboard or using APIs to transfer money to customers. You may also check our docs for detailed instructions.';
+    const button2 = 'How can I setup QR Codes?';
+    const description2 =
+      'Just use Razorpay. You may also check our docs for detailed instructions. Please use the search functionality to ask your queries.';
+    const externalButton1 = 'Expand First';
+    const externalButton2 = 'Expand Second';
+    const externalButtonCollapse = 'Collapse';
+
+    const ControlledAccordionExample = (): JSX.Element => {
+      const [expandedIndex, setExpandedIndex] = useState(-1);
+
+      return (
+        <>
+          <Button onClick={() => setExpandedIndex(0)}>{externalButton1}</Button>
+          <Button onClick={() => setExpandedIndex(1)}>{externalButton2}</Button>
+          <Button onClick={() => setExpandedIndex(2)}>Expand Third</Button>
+          <Button onClick={() => setExpandedIndex(-1)}>{externalButtonCollapse}</Button>
+          <Accordion
+            expandedIndex={expandedIndex}
+            onExpandChange={({ expandedIndex }) => setExpandedIndex(expandedIndex)}
+          >
+            <AccordionItem title={button1} description={description1} />
+            <AccordionItem title={button2} description={description2} />
+            <AccordionItem
+              title="How can I setup Subscriptions?"
+              description="Just use Razorpay."
+            />
+          </Accordion>
+          ,
+        </>
+      );
+    };
+
+    const { getByText, getAllByRole } = renderWithTheme(<ControlledAccordionExample />);
+
+    /**
+     * - Button 1 -> External Button 1
+     * - Button 2 -> External Button 2
+     * - Button 3 -> External Button 3
+     * - Button 4 -> External Button Collapse
+     * - Button 5 -> AccordionItem button 1
+     * - Button 6 -> AccordionItem button 2
```

Comment:
😅 kind of hard to track via indices, can we save them in variable and reference these?

Reply:
Done in https://github.com/razorpay/blade/commit/5362504ae49ec5bbbae577c5543de2e9ba15b3cc